### PR TITLE
refactor: Remove deprecated logging types

### PR DESCRIPTION
**Pull request by alexvy86**
## Description

Removes several telemetry-related types that had been deprecated in the past.

- In telemetry-utils
  - `ITelemetryErrorEvent`
  - `ITelemetryGenericEvent`
  - `ITelemetryPerformanceEvent`
  - `TelemetryNullLogger`
- In common-utils:
  - `BaseTelemetryNullLogger`
  - `TelemetryNullLogger`

Internally replaced uses of `ITelemetryErrorEvent`, `ITelemetryGenericEvent`, and `ITelemetryPerformanceEvent`, with their `*Ext` counterparts.

## Breaking Changes

The types called out above as being removed.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

The changes to common-utils would not cause an immediate break of the build if they were being used somewhere because this change doesn't update the dependency version in packages that consume common-utils, but there are no matches in the repo for either `TelemetryNullLogger` or `BaseTelemetryNullLogger`.

I did not add a changeset for the common-utils removals. They are marked @internal so in practice they'll just disappear from the 2.x versions of the package along with many other things.... I'm not sure if we want independent changesets for every API that's going away in 2.x because it became internal.

### CODE REVIEW

File: packages/loader/container-loader/src/deltaManager.ts [hunk 1]
```diff
@@ -376,7 +376,7 @@ export class DeltaManager<TConnectionManager extends IConnectionManager>
 	 * we stop processing ops that results in no processing join op and thus moving to connected state)
 	 * @param event - Event to log.
 	 */
-	public logConnectionIssue(event: ITelemetryErrorEvent) {
+	public logConnectionIssue(event: ITelemetryErrorEventExt) {
```
Comment (alexvy86) [inline packages/loader/container-loader/src/deltaManager.ts:379]: This only exists on `DeltaManager` the class, which isn't exported (as opposed to `IDeltaManager` the interface).

File: .changeset/tired-pianos-battle.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+Removed deprecated telemetry event types
+
+The deprecated `ITelemetryErrorEvent`, `ITelemetryGenericEvent`, and `ITelemetryPerformanceEvent` interfaces,
+which represented different kinds of telemetry events, were not intended for consumers of Fluid Framework and have thus
+been removed.
+`ITelemetryBaseEvent` is the only telemetry event interface that should be used in/by consuming code.
+
+`ITelemetryLogger` was not intended for consumers of Fluid Framework and has been removed.
+Consumers should use the simpler `ITelemetryBaseLogger` instead.
```
Comment (Josmithr) [inline .changeset/tired-pianos-battle.md:13]: Not for this PR, but we should consider renaming the type we expose to customers to be simpler. IMO, the thing we export should be `ITelemetryLogger`, and whatever internal type we use should be named something more explicit.

File: packages/loader/driver-utils/api-report/driver-utils.api.md [hunk 3]
```diff
@@ -30,7 +30,7 @@ import { IStreamResult } from '@fluidframework/driver-definitions';
 import { ISummaryContext } from '@fluidframework/driver-definitions';
 import { ISummaryHandle } from '@fluidframework/protocol-definitions';
 import { ISummaryTree } from '@fluidframework/protocol-definitions';
-import { ITelemetryErrorEvent } from '@fluidframework/core-interfaces';
+import { ITelemetryErrorEventExt } from '@fluidframework/telemetry-utils';
```
Comment (Josmithr) [inline packages/loader/driver-utils/api-report/driver-utils.api.md:33]: Presumably we'll be able to rename all of these "Ext" types, now that their corresponding base interfaces are gone? Subsequent PR?

**Review by Josmithr — APPROVED** [review]
🥳

File: .changeset/tired-pianos-battle.md (hunk 2 above)
Comment (alexvy86) [reply .changeset/tired-pianos-battle.md:13]: I like that idea. @markfields ?

File: packages/loader/driver-utils/api-report/driver-utils.api.md (hunk 3 above)
Comment (alexvy86) [reply packages/loader/driver-utils/api-report/driver-utils.api.md:33]: Yeah, I wasn't thinking about that step yet but definitely can be done in a subsequent PR.

**Pull request merged**